app/controllers/services_controller.rb

```diff
@@ -68,6 +66,24 @@ def deactivate
     end
   end
 
+  def public_set
+    @service = Service.find_by_id(params[:id])
```

Comment:
Use the before action :loadservice_and_authorize for all actions that need find_by_id